### PR TITLE
[helm] add local k3d loki cluster

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

This removes the `loki-distributed` k3d cluster, and adds on OSS `loki` cluster that is installed using `helm install`. This follows the pattern used by the `enterprise-logs` cluster, and both clusters share the same base `helm-cluster` jsonnet configuration.

To spin up a local Loki cluster, run `make k3d-loki` from the root of the repo.
To spin up a local GEL cluster, run `make k3d-enterprise-logs` from the root of the repo.
To tear either cluster down, run `make k3d-down` from the root of the repo.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Comment by trevorwhitney** [comment]
cc @jeschkies @ssncferreira as I know you both have used these k3d envs for GEL

File: Makefile [hunk 1]
```diff
@@ -773,3 +774,12 @@ check-example-config-doc: generate-example-config-doc
 		echo -e "(Don't forget to check in the generated files when finished)\n"; \
 		exit 1; \
 	fi
+
+k3d-loki:
```
Comment (jeschkies) [inline Makefile:778]: The name is a little confusing. How about `tanka-loki` instead?

**Review by jeschkies — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (trevorwhitney) [reply Makefile:778]: maybe `helm-loki-cluster`? it doesn't use `tanka` to install Loki, so I think `tanka-loki` is also misleading.

Comment (trevorwhitney) [reply Makefile:778]: or `k3d-loki-cluster`? It spins up a `k3d` cluster using `tanka`, which it then `helm installs` Loki into.

Comment (jeschkies) [reply Makefile:778]: How about `dev-k3d-...`

**Review by jeschkies — APPROVED** [review]

**Pull request merged**